### PR TITLE
Add section component

### PR DESCRIPTION
**Pull request by justinvdm**
These are used in multiple places in the app, including the category about page and call notes.

### CODE REVIEW

File: src/components/Paragraph/__tests__/Paragraph-test.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  it('should render', () => {
+    expect(render(
+      <Paragraph>
+        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+        eiusmod tempor incididuntut labore et dolore magna aliqua. Ut enim ad
+        minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+        aliquip ex ea commodo consequat
+      </Paragraph>
+    )).toMatchSnapshot();
+  });
```
Comment (justinvdm) [inline src/components/Paragraph/__tests__/Paragraph-test.js]: screen shot 2016-08-25 at 10 51 27 am

File: src/components/Paragraph/__tests__/Paragraph-test.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+describe('Paragraph', () => {
+  it('should render', () => {
+    expect(render(
+      <Paragraph>
+        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+        eiusmod tempor incididuntut labore et dolore magna aliqua. Ut enim ad
+        minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+        aliquip ex ea commodo consequat
+      </Paragraph>
+    )).toMatchSnapshot();
```
Comment (justinvdm) [inline src/components/Paragraph/__tests__/Paragraph-test.js]: screen shot 2016-08-25 at 10 53 50 am

File: src/components/Paragraph/__tests__/Paragraph-test.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  it('should support style overrides', () => {
+    expect(render(
+      <Paragraph style={{ color: '#111' }}>
+        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+        eiusmod tempor incididuntut labore et dolore magna aliqua. Ut enim ad
+        minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+        aliquip ex ea commodo consequat
+      </Paragraph>
+    )).toMatchSnapshot();
```
Comment (justinvdm) [inline src/components/Paragraph/__tests__/Paragraph-test.js]: screen shot 2016-08-25 at 10 56 12 am

File: src/components/Section/__tests__/Section-test.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+describe('Section', () => {
+  it('should render', () => {
+    expect(render(
+      <Section title="Foo bar baz">
+        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
+        eiusmod tempor incididuntut labore et dolore magna aliqua. Ut enim ad
+        minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+        aliquip ex ea commodo consequat
+      </Section>
+    )).toMatchSnapshot();
```
Comment (justinvdm) [inline src/components/Section/__tests__/Section-test.js]: screen shot 2016-08-25 at 10 51 27 am

File: src/components/SectionTitle/__tests__/SectionTitle-test.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { SectionTitle } from 'src/components';
+
+
+describe('SectionTitle', () => {
+  it('should render', () => {
+    expect(render(
+      <SectionTitle>Lorem ipsum dolor sit amet</SectionTitle>
+    )).toMatchSnapshot();
```
Comment (justinvdm) [inline src/components/SectionTitle/__tests__/SectionTitle-test.js]: screen shot 2016-08-25 at 11 00 25 am

File: src/components/SectionTitle/__tests__/SectionTitle-test.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  });
+
+  it('should support style overrides', () => {
+    expect(render(
+      <SectionTitle style={{ color: '#333' }}>
+        Lorem ipsum dolor sit amet
+      </SectionTitle>
+    )).toMatchSnapshot();
```
Comment (justinvdm) [inline src/components/SectionTitle/__tests__/SectionTitle-test.js]: screen shot 2016-08-25 at 12 40 34 pm

**Comment by justinvdm** [comment]
@peterp ready for review

**Comment by peterp** [comment]
👍 

I'm a bit cautious about `Paragraph` as it doesn't do much different from text (besides some styling.)

But... I don't think we have a need for it anywhere else other than what you've highlighted, so it should be OK. 🍰

File: src/components/Text/transform.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  const textTransform = getTextTransform(style, transforms);
+
+  switch (textTransform) {
+    case 'uppercase':
+      return text.toUpperCase();
+
+    default:
+      return text;
+  }
+};
```
Comment (justinvdm) [inline src/components/Text/transform.js]: So, I know how this looks. The alternatives aren't great though:
  - having components for each text style that needs uppercase
  - separating text components into uppercase text components and lowercase text components
  - using a `type` property on `Text` instead (means we can't use the nice `<Text styles={Text.types.paragraph}>` way of specifying types, and will also involve looking up similar to here)
  - having to specify `uppercase` on text explicitly (these should be tied to the text type, and we may accidentally omit it from one instance of a text type)
 
Unless we are doing really weird things, I don't see any performance issues with this approach. It does involve a bit of magic, but I think it is better than the alternatives.

**Comment by justinvdm** [comment]
screen shot 2016-08-25 at 7 35 25 pm

**Comment by justinvdm** [comment]
@peterp ready for review

**Comment by justinvdm** [comment]
screen shot 2016-08-25 at 7 31 50 pm

**Comment by justinvdm** [comment]
@peterp ready for review

**Comment by peterp** [comment]
👍

**Pull request merged**